source

```diff
@@ -88993,45 +88999,55 @@ dictionary <dfn>PromiseRejectionEventInit</dfn> : <span>EventInit</span> {
-       context</span> is not able to sustain this rate, it might drop to a more sustainable 30Hz for
-       that set of <code>Document</code>s, rather than occasionally dropping frames. (This
-       specification does not mandate any particular model for when to update the rendering.)
-       Similarly, if a <span>top-level browsing context</span> is in the background, the user agent
-       might decide to drop that page to a much slower 4Hz, or even less.</p>
-
-       <p>Another example of why a browser might skip updating the rendering is to ensure certain
-       <span data-x="concept-task">tasks</span> are executed immediately after each other, with only
-       <span data-x="perform a microtask checkpoint">microtask checkpoints</span> interleaved (and
-       without, e.g., <span data-x="run the animation frame callbacks">animation frame
-       callbacks</span> interleaved). For example, a user agent might wish to coalesce timer
-       callbacks together, with no intermediate rendering updates.</p>
-      </div>
+      <p>If there are a <span data-x="browsing context">browsing contexts</span> <var>B</var> that
+      do not have a rendering opportunity, then remove from <var>docs</var> all
+      <code>Document</code> objects whose <span data-x="concept-document-bc">browsing context</span>
+      is in <var>B</var>.</p>
+
+      <p>A <span>browsing context</span> has a rendering opportunity if the user agent is currently
+      able to present the contents of the <span>browsing context</span> to the user, accounting for
+      hardware refresh rate constraints and user agent throttling for performance reasons, but
+      considering content presented even if it's outside the viewport.</p>
+
+      <p class="note"><span>Browsing context</span> rendering opportunities are determined based on
+      hardware constraints such as display refresh rates and other factors such as page performance
+      or whether the page is in the background. Rendering opportunities typically occur at regular
+      intervals. For example, if the browser is attempting to achieve a 60Hz refresh rate, then
+      rendering opportunities occur at a maximum of every 60th of a second (about 16.7ms). If the
+      browser finds that a <span>browsing context</span> is not able to sustain this rate, it might
+      drop to a more sustainable 30 rendering opportunities per second for that <span>browsing
+      context</span>, rather than occasionally dropping frames. (This specification does not mandate
+      any particular model for selecting rendering opportunities.)  Similarly, if a <span>browsing
+      context</span> is not visible, the user agent might decide to drop that page to a much slower
+      4 rendering opportunities per second, or even less.</p>
      </li>
 
      <li>
-      <p>If there are a <span data-x="nested browsing context">nested browsing contexts</span>
```

Comment:
I'm not sure why we previously handled nested and top level browsing contexts independently. This seems clearer to me.

Reply:
It may have been incrementally added, and nobody thought to merge them.